test/test_futures.py

```diff
@@ -0,0 +1,101 @@
+import threading
+import time
+import torch
+from torch.futures import Future
+from torch.testing._internal.common_utils import TestCase, TemporaryFileName
+
+
+def add_one(fut):
+    return fut.wait() + 1
+
+
+class TestFuture(TestCase):
+    def test_wait(self):
+        f = Future()
+        f.set_result(torch.ones(2, 2))
+
+        self.assertEqual(f.wait(), torch.ones(2, 2))
+
+    def test_wait_multi_thread(self):
+
+        def slow_set_future(fut, value):
+            time.sleep(0.5)
+            fut.set_result(value)
+
+        f = Future()
+
+        t = threading.Thread(target=slow_set_future, args=(f, torch.ones(2, 2)))
+        t.daemon = True
```

Comment:
Curious, why set this as a daemon instead of joining after the assertEqual?

Reply:
it should be joined, my bad.